src/mgmt/cli_mgmt.c

```diff
@@ -583,41 +578,37 @@ int dc_mgmt_net_cfg(const char *name)
 
 	/* Allow client env overrides for these three */
 	d_agetenv_str(&crt_timeout, "CRT_TIMEOUT");
-	if (!crt_timeout) {
-		rc = asprintf(&crt_timeout, "%d", info->crt_timeout);
-		if (rc < 0) {
-			crt_timeout = NULL;
-			D_GOTO(cleanup, rc = -DER_NOMEM);
-		}
-		D_INFO("setenv CRT_TIMEOUT=%s\n", crt_timeout);
-		rc = d_setenv("CRT_TIMEOUT", crt_timeout, 1);
-		if (rc != 0)
-			D_GOTO(cleanup, rc = d_errno2der(errno));
-	} else {
+	if (!crt_timeout)
+		crt_info->cio_crt_timeout = info->crt_timeout;
+	else
 		D_DEBUG(DB_MGMT, "Using client provided CRT_TIMEOUT: %s\n", crt_timeout);
-	}
-
-	/* client-provided iface/domain were already taken into account by agent */
-	/* TODO: These should be set in crt_init_options_t instead of env */
-	rc = d_setenv("D_INTERFACE", info->interface, 1);
-	if (rc != 0)
-		D_GOTO(cleanup, rc = d_errno2der(errno));
-
-	rc = d_setenv("D_DOMAIN", info->domain, 1);
-	if (rc != 0)
-		D_GOTO(cleanup, rc = d_errno2der(errno));
 
 	sprintf(buf, "%d", info->provider_idx);
 	rc = d_setenv("CRT_SECONDARY_PROVIDER", buf, 1);
 	if (rc != 0)
 		D_GOTO(cleanup, rc = d_errno2der(errno));
 
+	D_STRNDUP(crt_info->cio_provider, info->provider, DAOS_SYS_INFO_STRING_MAX);
+	if (NULL == crt_info->cio_provider)
+		D_GOTO(cleanup, rc = -DER_NOMEM);
+	D_STRNDUP(crt_info->cio_interface, info->interface, DAOS_SYS_INFO_STRING_MAX);
+	if (NULL == crt_info->cio_interface) {
+		D_FREE(crt_info->cio_provider);
+		D_GOTO(cleanup, rc = -DER_NOMEM);
+	}
+	D_STRNDUP(crt_info->cio_domain, info->domain, DAOS_SYS_INFO_STRING_MAX);
+	if (NULL == crt_info->cio_domain) {
+		D_FREE(crt_info->cio_provider);
+		D_FREE(crt_info->cio_interface);
+		D_GOTO(cleanup, rc = -DER_NOMEM);
+	}
+
 	D_INFO("Network interface: %s, Domain: %s\n", info->interface, info->domain);
 	D_DEBUG(DB_MGMT,
 		"CaRT initialization with:\n"
 		"\tD_PROVIDER: %s, CRT_TIMEOUT: %s, "
 		"CRT_SECONDARY_PROVIDER: %s\n",
-		provider, crt_timeout, buf);
+		crt_info->cio_provider, crt_timeout, buf);
```

Comment:
crt_timeout print here is misleading as it could be info->crt_timeout instead if set. See comment on line 597

Reply:
fixed